src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -124,6 +124,7 @@ public Task<OidcMetadata> DiscoverOidcMetadataAsync(Uri endpoint, RequestContext
                         {
                             var requestData = new OnBeforeTokenRequestData(_bodyParameters, _headers, endpointUri, requestContext.UserCancellationToken);
                             await onBeforePostRequestData(requestData).ConfigureAwait(false);
+                            endpointUri = requestData.RequestUri;
```

Comment:
What about a null check since RequestUri has a public setter (body params and headers properties don't).

```suggestion
                            if (requestData.RequestUri != null)
                            {
                                endpointUri = requestData.RequestUri;
                            }
```

Reply:
MSAL guarantees that the `request.RequestUri` is populated. If the app developers sets it to null, I saw we obey their wish :) - it's an extensibility  API after all.

Eventually the endpoint is used with HttpClient APIs like `HttpRequestMessage requestMessage = new HttpRequestMessage { RequestUri = endpoint };` so eventually app dev will get an error from the Http stack.